### DISCUSSION TITLE
有关如何找到清单、密钥并分享至讨论区

### DISCUSSION THREAD

**Discussion by Nobody2613** [pjy612/SteamManifestCache#515] [Question / Answer] [answered]
如题

**Comment by Nobody2613** [comment, marked as answer]
### 关于如何找到相关游戏的清单与密钥：

- **密钥**：可于该文件"C:\Program Files (x86)\Steam\config\config.vdf"中获取
	- 该目录下.vdf文件的打开方式：使用一般的文本编辑器打开
	（例如Windows系统自带的记事本，或Notepad++，可于[久治明千树汐 / Notepad++ · GitLab (soraharu.com)](https://gitlab.soraharu.com/XiaoXi/Notepad-Plus-Plus/-/tree/master)下载）

	- 在config.vdf文件中寻找密钥：
	使用编辑器功能“查找”搜索关键词“depots”或“DecryptionKey”，如果您搜索到了形如下列样式的一段文本，那么说明您找到了正确的位置
	```
	"depots"
	{
	   "228983"
	   {
		   "DecryptionKey"		   "77c8e812cd79e67e2d376721253ebb07e06b3646f05671c6c9517b27be14734b"
	   }
	......
	```
	- 找到确定游戏的对应密钥：
	1. 获取到对应游戏文件的depotid：

		1. 在[SteamDB](https://steamdb.info/)上搜索对应游戏的名字
		2. 点进对应游戏的页面，例如动物园之星的信息页面为[Planet Zoo · SteamDB](https://steamdb.info/app/703080/)
		3. 切换到Depots栏，查看游戏具体的depotid，一般为拥有较大体积的或明确标识了Content、DLC字样的depot，例如动物园之星的depot页面[Planet Zoo Depots · SteamDB](https://steamdb.info/app/703080/depots/) 中703081、1098120分别是游戏本体与dlc的depotid

	2. 在config.vdf的对应区域中搜索depotid，并将对应的DecryptionKey记录以特定格式记录下来，以太吾绘卷的本体838351为例
		- `838351;7de3577779d63fd52db715a1898f8353eb0f5f4096ab3391f77b582621e1adb8`
		可以发现，这样的特定格式其实就是形如`<depotid>;<DecryptionKey>`的格式
	
	
	- 批量格式转换可以参考：[#60](https://github.com/pjy612/SteamManifestCache/discussions/60)

- **清单**：可于"C:\Program Files (x86)\Steam\depotcache"目录中获取
	- 清单文件的文件名格式为`<depotid>_<manifestid>.manifest`
	
	- 获取Manifest ID
		以太吾绘卷为例，在SteamDB太吾绘卷的Depot页面中，点击对应的depotid可进入该depot的信息页面
		例如depotid为838351的depot
			可以在该页面[Depot 838351 (The Scroll Of Taiwu Alpha) · SteamDB](https://steamdb.info/depot/838351/) 看到其 Manifest ID 为7533706495254451196
		那么对应的，其清单文件的文件名应为`838351_7533706495254451196.manifest`

- **一些小问题**：如果在config.vdf或depotcache文件夹中找不到相关的密钥或清单文件，亦或是清单id与SteamDB上的不相符，请考虑：
	- 更新游戏
	- 下载游戏（您不需要真正的下载游戏，在游戏已经有下载哪怕1B数据的时候，清单文件或密钥已经在对应位置生成了，在完成密钥与清单文件的备份后可以卸载正在下载中的游戏）
	- 验证游戏完整性
	

### 关于如何将相关信息上传至分享板块：

- **格式**：具体的格式信息已几乎详尽的由pjy612大佬写在[分享模板](https://github.com/pjy612/SteamManifestCache/discussions/26)中
	- 例：
		``````
		[appid] 游戏名/软件名 | 本体 DLC（如果有的话）
		`appid:<appid>`
		keys:
		```
		<depotid>;<DecryptionKey>
		```
		email;邮箱
		``````
	- 实际案例：
		- 本体+DLC
		``````
		[1111460] Bunny eShop | 本体 DLC
		`appid:1111460`
		keys:
		```
		1111461;9d700b248094587daf680af4ab5b65bb8506f3e53bfea3693bcd9c528a336957
		1796170;6b1c0481f79cf61f11536b19682613eb52e87e3f726e53b524e81730f5b0bbd6
		2274630;5785cc9c6864b561f27b0600d9befa86fa4b10df8af0f8ee177737cbf61ce2b3
		```
		email;nobody53_@outlook.com
		``````
		- 仅本体
		``````
		[629520] Soundpad | 本体
		`appid:629520`
		keys:
		```
		629521;40df4fcb0748caf1d77bff732f597eebb26b3a36539b1646d149f54eb676807f
		```
		email;nobody53_@outlook.com
		``````
	- 注意事项：
		在获取到相关资格后无需再附带邮箱

- **上传清单文件**：最好将清单文件压缩为压缩包，然后在上述文字信息输入完毕后另起一行，在附件中添加该压缩包，有关如何在github评论中添加附件，请参考：[附加文件 - GitHub 文档](https://docs.github.com/zh/get-started/writing-on-github/working-with-advanced-formatting/attaching-files)
- **没什么问题的分享贴，可供参考**：
[格式正确的分享](https://github.com/pjy612/SteamManifestCache/discussions?discussions_q=is%3Aopen+label%3A%22good+share%22)
[[881020] 碧蓝幻想Relink](https://github.com/pjy612/SteamManifestCache/discussions/445)
[[1584090]东方夜雀食堂 - Touhou Mystia's Izakaya{本体+6DLC+OST}](https://github.com/pjy612/SteamManifestCache/discussions/496)

### 关于其他问题或名词/原理解释：
- 请参考[pjy612/SteamManifestCache Wiki (github.com)](https://github.com/pjy612/SteamManifestCache/wiki)

### 关于如何在讨论区发布帖子以进行分享：
- 请参考[Quickstart for GitHub Discussions - GitHub Docs](https://docs.github.com/en/discussions/quickstart#creating-a-new-discussion)
	- 需要注意：在讨论类别中选择密钥清单分享

**Comment by AT7921** [comment]
内测码在哪看

**Reply by Nobody2613** [reply]
可以尝试使用SAI解锁一个App来生成内测码

**Reply by AT7921** [reply]
我没有sai

**Reply by Nobody2613** [reply]
那就不填（

**Comment by Gaoway** [comment]
没内测码，求下载sai的方法

**Reply by Nobody2613** [reply]
不填内测码分享试试吧

**Reply by pjy612** [reply]
sai 已停止维护

**Comment by 1275688416** [comment]
lua文件是怎么搞的啊

**Reply by pjy612** [reply]
用手写

**Reply by Fuyuppe** [reply]
any tutorial for this?

**Reply by pjy612** [reply]
> any tutorial for this?

Copying
infer other things from one fact

**Reply by Fuyuppe** [reply]
addappid(2091550)
addappid(2091551,1,"b6863c86e01e38ffc8fd3e4370e9f503e3bff12fdb0f78a2e9c34cf679566419")
setManifestid(2091551,"8047248613977667084",6013986595)
2091550 =app id
2091551 = depot
1 = ?
b6863c86e01e38ffc8fd3e4370e9f503e3bff12fdb0f78a2e9c34cf679566419 = decryption key
8047248613977667084 = manifest id
6013986595 = ?
Could you explain to me '?' that I'm still not understanding, please?
 1 = ? (how could I get this one from)
6013986595 = ? (this one too) 
Thank you in advance

**Reply by pjy612** [reply]
addappid( appid )
addappid( depotid ,1, depotkey )

setManifestid( depotid , manifestid , size) size can set 0
